xbmc/cores/VideoRenderers/BaseRenderer.cpp

```diff
@@ -222,11 +222,20 @@ void CBaseRenderer::FindResolutionFromFpsMatch(float fps, float& weight)
 RESOLUTION CBaseRenderer::FindClosestResolution(float fps, float multiplier, RESOLUTION current, float& weight)
 {
   RESOLUTION_INFO curr = g_graphicsContext.GetResInfo(current);
+  RESOLUTION orig_res  = CDisplaySettings::GetInstance().GetCurrentResolution();
+
+  if (orig_res <= RES_DESKTOP)
+    orig_res = RES_DESKTOP;
+
+  RESOLUTION_INFO orig = g_graphicsContext.GetResInfo(orig_res);
 
   float fRefreshRate = fps;
 
   float last_diff = fRefreshRate;
 
+  int curr_diff = std::abs((int) m_sourceWidth - curr.iScreenWidth);
```

Comment:
nope iScreenWidth is an int: https://github.com/xbmc/xbmc/blob/master/xbmc/guilib/Resolution.h#L92 so I only take care that m_sourceWidth which is unsigned int (but in a range int can manage) is casted first, so that the abs should use the int prototype.